### ISSUE TITLE
StreamContextVariable's create_proxy logic removed in Dynamo post-2.9, breaking custom backends relying on set_stream nodes

### ISSUE THREAD

**Issue by BobBob-Fu** [pytorch/pytorch#176154]
I observed that in PyTorch 2.10, the enter and exit methods of StreamContextVariablein the Dynamo component have removed the logic for creating proxies via tx.output.create_proxy. In contrast, PyTorch 2.9 retains this logic:
```
    def enter(self, tx):
        # stream generated inside the traced function
        if self.target_values[0].as_proxy() is not None:
            tx.output.create_proxy(
                "call_function",
                self.set_stream,
                (self.target_values[0].as_proxy(),),
                {},
            )
```
This change causes the set_stream nodes to disappear from FX graphs in PyTorch 2.10+, whereas they were present in 2.9.

related pr maybe: https://github.com/pytorch/pytorch/pull/166472

**Example**
```
import torch

def my_backend(gm, example_inputs):
    print(gm.graph)
    return gm

@torch.compile(backend=my_backend)
def cus_func(a):
    s = torch.cuda.Stream()
    with torch.cuda.stream(s):
        torch.add(a, 1)

a = torch.ones(2, 2).cuda()
cus_func(a)
```
In torch2.9, FX graph will be:
```
graph():
    %l_a_ : torch.Tensor [num_users=1] = placeholder[target=L_a_]
    %stream : [num_users=1] = call_function[target=torch.cuda.streams.Stream](args = (), kwargs = {})
    %current_stream : [num_users=1] = call_function[target=torch.cuda.current_stream](args = (None,), kwargs = {})
    %set_stream : [num_users=0] = call_function[target=torch.cuda.set_stream](args = (%stream,), kwargs = {})
    %add : [num_users=0] = call_function[target=torch.add](args = (%l_a_, 1), kwargs = {})
    %set_stream_1 : [num_users=0] = call_function[target=torch.cuda.set_stream](args = (%current_stream,), kwargs = {})
    return ()
```

**Impact:​**
My custom torch.compilebackend relies on these set_stream nodes in the FX graph to perform stream-related operations before AOTAutograd. The removal of these nodes breaks my workflow, as the necessary stream configuration is no longer captured in the graph.

**Questions:**
* What motivated the removal of create_proxyin StreamContextVariable?
* Are there alternative approaches to retain the original behavior (e.g., explicit stream management) in newer Dynamo versions?

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @kadeng @amjames @Lucaskabela @jataylo @mlazos

**Comment by mlazos** [comment]
The motivation is that the old streams handling was very broken and incomplete. We now have full feature support for it.

Compiling streams is available in the nightly, let me know if you have any issues with it. The way it works now is that we will annotate nodes with their stream in the metadata, and also include event.[record, wait, syncrhonize] in the traced graph. Let me know if you have any issues or questions with it. Tag me and apply the  label "module: compiled streams" to any issues you encounter.
